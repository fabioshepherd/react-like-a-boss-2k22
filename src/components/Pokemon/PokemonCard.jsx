import { Grid, Typography } from '@mui/material';

const PokemonCard = ({ number, name, onClick }) => {
  return (
    <Grid
      container
      direction='column'
      alignItems={'center'}
      sx={{
        padding: '16px 8px',
        boxShadow: '0px 0px 21px 0px rgba(0,0,0,0.1)',
        borderRadius: '16px',
      }}
      onClick={onClick}
    >
      <Grid item xs='auto' sx={{ marginBottom: '8px' }}>
        <Typography variant='h5'>{number}</Typography>
      </Grid>
      <Grid item>
        <Typography variant='h6' sx={{ textTransform: 'capitalize' }}>
          {name}
        </Typography>
      </Grid>
    </Grid>
  );
};

export default PokemonCard;
