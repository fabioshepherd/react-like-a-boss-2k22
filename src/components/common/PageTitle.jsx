import styled from '@emotion/styled';
import { Typography } from '@mui/material';

const PageTitleDiv = styled.div(() => ({
  padding: '16px 0',
  marginBottom: '24px',
}));

const PageTitle = ({ title }) => {
  return (
    <PageTitleDiv>
      <Typography variant={'h3'}>{title}</Typography>
    </PageTitleDiv>
  );
};

export default PageTitle;
