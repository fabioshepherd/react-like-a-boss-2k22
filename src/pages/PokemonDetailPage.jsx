import styled from '@emotion/styled';
import { Grid } from '@mui/material';
import { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import PageTitle from '../components/common/PageTitle';
import { getPokemon, getPokemonByNumber } from '../services/pokemonService';

const PokemonDetailPage = () => {
  const { number } = useParams();

  const [pokemon, setPokemon] = useState();

  useEffect(() => {
    prepareData();
  }, []);

  const prepareData = async () => {
    try {
      const pokemonResponse = await getPokemonByNumber(number);

      setPokemon(pokemonResponse.data);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <>
      <PageTitle title={'Pokemon!'} />
      <Grid container spacing={3} justifyContent='center'></Grid>
    </>
  );
};

export default PokemonDetailPage;
