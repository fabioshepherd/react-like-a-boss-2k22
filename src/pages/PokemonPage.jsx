import styled from '@emotion/styled';
import { Grid } from '@mui/material';
import { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import PageTitle from '../components/common/PageTitle';
import PokemonCard from '../components/Pokemon/PokemonCard';
import { getPokemonList } from '../services/pokemonService';

const PokemonPage = () => {
  const [pokemons, setPokemons] = useState([]);
  const navigate = useNavigate();

  useEffect(() => {
    prepareData();
  }, []);

  const prepareData = async () => {
    try {
      const limit = 400;
      const pokemonResponse = await getPokemonList(limit);

      setPokemons(pokemonResponse.data?.results);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <>
      <PageTitle title={'Pokemon!'} />
      <Grid container spacing={3} justifyContent='center'>
        {pokemons.length > 0 &&
          pokemons.map((pokemon, i) => (
            <Grid item xs={4} sm={3} key={i}>
              <PokemonCard
                name={pokemon.name}
                number={i + 1}
                key={i}
                onClick={() => {
                  navigate(`${i * 1 + 1}`);
                }}
              />
            </Grid>
          ))}
      </Grid>
    </>
  );
};

export default PokemonPage;
