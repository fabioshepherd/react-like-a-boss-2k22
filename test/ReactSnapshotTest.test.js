import oo,{ MemoryRouter, Route, Routes } from "react-router-dom";
import ReactRouter  from "react-router";
import renderer from "react-test-renderer";
import PageTitle from "../src/components/common/PageTitle";
import PokemonCard from "../src/components/Pokemon/PokemonCard";
import PokemonDetailPage from "../src/pages/PokemonDetailPage";
import PokemonPage from "../src/pages/PokemonPage";
import { screen } from "@testing-library/react";
import { Timeline } from "@mui/icons-material";
import { setTimeout } from "timers/promises"

afterEach(() => {
  
 // jest.clearesetAllMocks();
})

beforeEach(() => {
  jest.mock("react-router-dom",() => ({
    ...jest.requireActual("react-router-dom"),
    useParams: () => ({
      number: "1"
    }),
    useRouteMatch: () => ({ url: '/pokemon/1' }),
  }))
})


describe("Snapshot Tests", () => { // descrive dopo --> all'inizio solo il test
  test("PageTitle Test", () => {
    const component = renderer.create(<PageTitle title={"Titolo pagina"} />);
    let jsonRap = component.toJSON();
    expect(jsonRap).toMatchSnapshot(); // h3 --> h4
  });

  test("PokemonCard Test", () => {
    const component = renderer.create(<PokemonCard number={10} name={"bulbasaur"} onClick={() => {}} />)
    const jsonRap = component.toJSON();
    expect(jsonRap).toMatchSnapshot();
  })

  test("PokemonPage Test", () => {
    //const history = createMemoryHistory();
    //const component = renderer.create(<Router location={history.location} navigator={history}><PokemonPage /></Router>);

    const component = renderer.create(<PokemonPage />);
    const jsonRap = component.toJSON();
    expect(jsonRap).toMatchSnapshot();
  })

  test("PokemonDetailPage Test", async () => {

   
    //jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ number: '1' });
    //console.log("useParams()");
   // console.log(oo.useParams());



    //let component = renderer.create(<MemoryRouter initialEntries={['/pokemon/1']}><Routes><Route path="/pokemon/:number" element={<PokemonDetailPage />}></Route></Routes></MemoryRouter>)
   // const jsonRap = component.toJSON();
    //expect(jsonRap).toMatchSnapshot(); // problema, quello che è stato messo nello snapshot
                                        // 2 problema, non ho un indirizzo
   
  // await 
   
  // await renderer.act(()=> component = renderer.create(<MemoryRouter initialEntries={['/pokemon/1']}><Routes><Route path="/pokemon/:number" element={<PokemonDetailPage />}></Route></Routes></MemoryRouter>) )
  //  const jsonActRap = component.toJSON();
  //  expect(jsonActRap).toMatchSnapshot(); // problema, quello che è stato messo nello snapshot

   // await renderer.act(()=> component.update(<MemoryRouter initialEntries={['/pokemon/1']}><Routes><Route path="/pokemon/:number" element={<PokemonDetailPage />}></Route></Routes></MemoryRouter>) )
   // const jsonUpdRap = component.toJSON();
   // expect(jsonUpdRap).toMatchSnapshot(); // problema, quello che è stato messo nello snapshot*/
 
   let root; 
   await renderer.act(async () => {
     root = renderer.create(<MemoryRouter initialEntries={['/pokemon/1']}><Routes><Route path="/pokemon/:number" element={<PokemonDetailPage />}></Route></Routes></MemoryRouter>)
     expect(root.toJSON()).toMatchSnapshot();

    await setTimeout(1000)

      expect(root.toJSON()).toMatchSnapshot();
    });

  
/*
   await renderer.act(() => {
    root.update(<MemoryRouter initialEntries={['/pokemon/1']}><Routes><Route path="/pokemon/:number" element={<PokemonDetailPage />}></Route></Routes></MemoryRouter>);
  })
  
  // asserzioni sulla root
  expect(root.toJSON()).toMatchSnapshot();*/
})
});
