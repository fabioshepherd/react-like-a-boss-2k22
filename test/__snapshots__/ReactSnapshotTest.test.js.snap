// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests PageTitle Test 1`] = `
<div
  className="css-cc5o4n"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
  >
    Titolo pagina
  </h3>
</div>
`;

exports[`Snapshot Tests PokemonCard Test 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-tfd2vm-MuiGrid-root"
  onClick={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-mhz0a0-MuiGrid-root"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      10
    </h5>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1hwz2fv-MuiTypography-root"
    >
      bulbasaur
    </h6>
  </div>
</div>
`;

exports[`Snapshot Tests PokemonDetailPage Test 1`] = `null`;

exports[`Snapshot Tests PokemonDetailPage Test 2`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1ws4nd0-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-153cd0s-MuiGrid-root"
  >
    <img
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-jy0p6y-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 pokemon-name css-5lbw0b-MuiTypography-root"
        >
          bulbasaur
        </h4>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault pokemon-type css-gac2fo-MuiChip-root"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
        >
          <span
            className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            grass
          </span>
        </div>
        <div
          className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault pokemon-type css-gac2fo-MuiChip-root"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
        >
          <span
            className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            poison
          </span>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium info-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="HeightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 6.99h3L12 3 8 6.99h3v10.02H8L12 21l4-3.99h-3z"
          />
        </svg>
         height:
        <span
          className="info"
        >
           
          70cm
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium info-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="ScaleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 11V8c4.56-.58 8-3.1 8-6H2c0 2.9 3.44 5.42 8 6v3c-3.68.73-8 3.61-8 11h6v-2H4.13c.93-6.83 6.65-7.2 7.87-7.2s6.94.37 7.87 7.2H16v2h6c0-7.39-4.32-10.27-8-11zm-2 11c-1.1 0-2-.9-2-2 0-.55.22-1.05.59-1.41C11.39 17.79 16 16 16 16s-1.79 4.61-2.59 5.41c-.36.37-.86.59-1.41.59z"
          />
        </svg>
         weight:
        <span
          className="info"
        >
           
          6.9kg
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <img
              className="css-kihw5q"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <img
              className="css-kihw5q"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
